kernel/uid_gid.c

```diff
@@ -6,32 +6,89 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <myst/eraise.h>
 #include <myst/kernel.h>
+#include <myst/mount.h>
 #include <myst/syscall.h>
 #include <myst/thread.h>
 
-/* Currently only supports a single mapping which is typically from enclave root
- * identity to the host application identity when executed. */
-static myst_host_enc_id_mapping host_enc_id_mapping;
+static myst_host_enc_uid_mapping uid_mappings[MAX_ID_MAPPINGS];
+static int num_uid_mappings;
+static myst_host_enc_gid_mapping gid_mappings[MAX_ID_MAPPINGS];
+static int num_gid_mappings;
 
-void myst_set_host_uid_gid_mapping(myst_host_enc_id_mapping id_mapping)
+void myst_set_host_uid_gid_mappings(
+    myst_host_enc_uid_gid_mappings* uid_gid_mappings)
 {
-    host_enc_id_mapping = id_mapping;
+    /* Copy over uid mappings */
+    {
+        num_uid_mappings = uid_gid_mappings->num_uid_mappings;
+        for (int i = 0; i < num_uid_mappings; i++)
```

Comment:
I think you can use direct assignment for array of structs. It will perform copy on each array element.

Reply:
Can't seem to get direct assignment working, the compiler complains about the array type not being a valid modifiable lvalue.

Reply:
Oh, I must have confused with another language.

Reply:
In that case, we can use memcpy.

Reply:
Changed array of structs copying to use memcpy.